packages/components/tests/integration/components/hds/tabs/index-test.js

```diff
@@ -3,232 +3,555 @@
-        <T.Tab>Two</T.Tab>
-        <T.Panel>Content 1</T.Panel>
-        <T.Panel>Content 2</T.Panel>
+        <T.Tab @isSelected={{true}} @id="ONE">One</T.Tab>
+        <T.Tab @isSelected={{true}} @id="TWO">Two</T.Tab>
+        <T.Panel data-test="panel-1">Content 1</T.Panel>
+        <T.Panel data-test="panel-2">Content 2</T.Panel>
       </Hds::Tabs>
     `);
+    // await pauseTest();
+    assert.throws(function () {
+      throw new Error(errorMessage);
+    });
+  });
+
+  test('it should throw an assertion @selecteTabIndex if out of bound for the array of tabs', async function (assert) {
+    const errorMessage =
+      '"Hds::Tabs" has tried to set the indicator for an element that doesn\'t exist (the value 99 of `this.selectedTabIndex` is out of bound for the array `this.tabNodes`, whose index range is [0-1])';
+    assert.expect(2);
+    setupOnerror(function (error) {
+      assert.strictEqual(error.message, `Assertion Failed: ${errorMessage}`);
+    });
+    await this.createTabs({ selectedTabIndex: 99 });
+    assert.throws(function () {
+      throw new Error(errorMessage);
+    });
+  });
+
+  // ===============================================================
+
+  // TAB OPTIONS
+
+  test('`Tab` should render an icon if @icon is defined', async function (assert) {
+    await this.createTabs({ iconTab1: 'info' });
+    assert.dom('.hds-tabs__tab-icon').exists();
+    assert.dom('.hds-tabs__tab-icon').hasAttribute('data-test-icon', 'info');
+  });
+
+  test('`Tab` should render a badge if @count is defined', async function (assert) {
+    await this.createTabs({ countTab1: '5' });
     assert.dom('.hds-tabs__tab-count').exists();
     assert.dom('.hds-tabs__tab-count').hasText('5');
   });
+
+  // ===============================================================
+
+  // INDICATOR
+
+  test('tab indicator should respond to tab clicks', async function (assert) {
+    await this.createTabs();
+    let tablistStyle = find('.hds-tabs__tablist').style;
+    assert.strictEqual(
+      tablistStyle['cssText'],
+      '--indicator-left-pos: 0px; --indicator-width: 51px;'
+    );
+    // select tab 2
+    await click('[data-test="tab-2"] .hds-tabs__tab-button');
+    assert.strictEqual(
+      tablistStyle['cssText'],
+      '--indicator-left-pos: 51px; --indicator-width: 51px;'
+    );
+  });
+
+  test('tab indicator should respond to content size changes', async function (assert) {
+    this.set('count', 5);
+    await render(hbs`
+      <Hds::Tabs id="test-tabs" as |T|>
+        <T.Tab data-test="tab-1" @count="{{this.count}}">One</T.Tab>
+        <T.Tab data-test="tab-2">Two</T.Tab>
+        <T.Panel data-test="panel-1">Content 1</T.Panel>
+        <T.Panel data-test="panel-2">Content 2</T.Panel>
+      </Hds::Tabs>
+    `);
+    let tablistStyle = find('.hds-tabs__tablist').style;
+    assert.strictEqual(
+      tablistStyle['cssText'],
+      '--indicator-left-pos: 0px; --indicator-width: 81px;'
+    );
```

Comment:
I'm thinking something like this could help us test an approximate match – we can abstract it into a small function
```suggestion
    let tablistStyle = find('.hds-tabs__tablist').style;
    let regex = /[a-z-]+ *:([^:;]+)px;/g;
    let indicatorLeftPos = regex.exec(tablistStyle['cssText'])[1];
    let indicatorWidth = regex.exec(tablistStyle['cssText'])[1];
    assert.true(Math.abs(0 - indicatorLeftPos) <= 1);
    assert.true(Math.abs(82 - indicatorWidth) <= 1);
```

Reply:
Yeah I tried your suggestion as first thing, but in some cases the difference is bigger (eg. `109` vs `112`) so I had to go full-in with a custom function that does more complex checks depending on different conditions/use cases (including the case where we're comparing `0` with `0` so we can't use a percentage)